### ISSUE TITLE
Map view annotations auto layout support

### ISSUE THREAD

**Issue by ittna** [mapbox/mapbox-maps-ios#2124]
## Environment

- Xcode version: 15.2
- iOS version: 17.2
- Devices affected: all
- Maps SDK Version: 11.1.0

## Observed behavior and steps to reproduce

Auto layout in view annotations seems to work correctly if you use topAnchor, bottomAnchor, leadingAnchor, trailingAnchor. As soon as you add constraint for widthAnchor or centerXAnchor it doesn't position view annotation correctly on map anymore.

## Expected behavior

Support all auto layout constraints.

**Comment by persidskiy** [comment]
@ittna Hi, can you please create a minimal demo project for this issue?

**Comment by ittna** [comment]
@persidskiy Hi, I made demo example here https://github.com/ittna/mapbox-maps-ios/tree/bug/autolayout in the https://github.com/ittna/mapbox-maps-ios/blob/bug/autolayout/Apps/Examples/Examples/All%20Examples/Annotations/ViewAnnotationBasicExample.swift

I tried to reproduce it first with your `AnnotationView` but there it seems to work correctly. I changed it to use my `AnimatedAnnotationView` which is closer to the annotation view I use in my project and with that it seems to reproduce. As far as I can see there is at least no missing constraints i.e. layout should be well defined. Weird thing is that they also move on map when you pinch zoom on the map. 

https://github.com/mapbox/mapbox-maps-ios/assets/2304597/e0ff97ae-9b74-415c-b013-d3f9d401547b
